engine/src/main/java/org/terasology/rendering/dag/RenderTaskListGenerator.java

```diff
@@ -54,20 +55,53 @@ private void logList(String title, List<?> list) {
     private void generateTaskList() {
```

Comment:
Well, I validated my version of the elimination in the tests and it was working with the issue that is related with the last node. I knew that it required a refactor (as mentioned in a TODO), but it was functionally. Nevertheless, I adapted your version with small changes and tests. It's ready for review.